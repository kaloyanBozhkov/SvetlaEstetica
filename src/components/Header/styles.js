export default {
    header: {
        color: '#4B4B4B',
        backgroundColor: 'white',
        height: '100%',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingLeft: {
            xs: '2',
            sm: '4',
            md: '8',
            mdX: '16'
        },
        paddingRight: {
            xs: '2',
            sm: '4',
            md: '8',
            mdX: '16'
        },
        boxShadow: '0px 0px 5px rgba(0, 0, 0, 0.25)',
    },
    logo: {
        flexDirection: 'column',
        height: '6xl',
        justifyContent: 'center',
        flex: {
            xs: '1',
            base: 'none'
        },
        alignItems: {
            xs: 'center',
            base: 'flex-start'
        }
    },
    logoHeading: {
        fontSize: {
            xs: 'lg',
            mdX: 'xl'
        },
        lineHeight: {
            xs: 'tighter',
            md: 'wider'
        }
    },
    logoSubHeading: {
        fontSize: {
            xs: 'xxs',
            md: 'xs',
            mdX: 'md'
        },
    },
    navMenu: {
        flexDirection: {
            xs: 'column',
            base: 'row'
        },
        fontSize: 'sm',
        justifyContent: {
            xs: 'flex-start',
            base: 'center'
        },
        alignItems: {
            xs: 'flex-start',
            base: 'unset'
        },
        flex: '1',
        position: {
            xs: 'fixed',
            base: 'relative',
        },
        top: '0',
        left: '0',
        bottom: '0',
        width: {
            xs: '30rem',
            base: 'auto'
        },
        height: {
            xs: '100vh',
            base: '100%'
        },
        zIndex: {
            xs: 'hide',
            base: 'base'
        },
        borderRight: {
            xs: '1px solid',
            base: 'none'
        },
        borderColor: 'background.200',
        transform: {
            xs: 'translateX(-100%)',
            base: 'none',
        },
        bg: {
            xs: 'white',
            base: 'unset'
        }
    },
    navBtn: {
        bg: 'unset',
        width: {
            xs: '100%',
            base: '13'
        },
        height: {
            xs: '4rem',
            base: '100%'
        },
        _hover: {
            bg: 'unset'
        },
        marginBottom: {
            xs: '2',
            base: '0'
        }
    },
    navHeading: {
        display: {
            xs: 'flex',
            base: 'none'
        },
        alignItems: 'center',
        justifyContent: 'center',
        alignSelf: 'stretch',
        marginLeft: '2',
        marginRight: '2',
        height: '24',
        borderBottom: '1px solid',
        borderColor: 'background.200',
        fontSize: 'tighter',
        marginBottom: '5',
    },
    iconButton: {
        width: {
            xs: '3',
            md: '4'
        },
        height: {
            xs: '3',
            md: '4'
        },
        bg: 'unset',
        fontSize: {
            xs: 'sm',
            md: 'md',
            lg: 'lg'
        },
        _hover: {
            bg: 'unset'
        }
    },
    accountBtn: {
        order: {
            xs: '-1',
            base: '0'
        }
    },
    menuBtn: {
        alignItems: 'center',
        position: 'relative',
        height: '4',
        width: '4',
        padding: '2',
        cursor: 'pointer',
        display: {
            xs: 'flex',
            base: 'none'
        },
        order: '-2',
        // pseudo handled with scss
    },
    middleMenuBar: {
        height: '1px',
        width: '100%',
        bg: 'background.800',
    }
}