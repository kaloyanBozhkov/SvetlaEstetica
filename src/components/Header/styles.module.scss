.menuBtn {
    position: relative;
    &::before, &::after {
        transition: .3s;
        transition-delay: .4s; // match open/close of menu
        content: '';
        position: absolute;
        height: 1px;
        left: 1rem;
        right: 1rem;
        background-color: #4B4B4B;
    }

    &::before {
        top: 1.25rem;
    }

    &::after {
        bottom: 1.25rem;
    }

    > div {
        opacity: 1;
        transition: .15s;
        transition-delay: .345s; // match open/close of menu
    }

    &[data-active="true"] {
        &::before {
            transform: rotate(45deg);
            top: 1.9rem;
        }
        &::after {
            transform: rotate(-45deg);
            bottom: 2rem;
        }
        > div {
            opacity: 0;
        }
    }
}

.wrapper {
    > * {
        margin: .5rem;
    }
}