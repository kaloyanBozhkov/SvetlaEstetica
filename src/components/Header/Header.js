import React from 'react'
import { Box, Button, Flex, Heading, Text } from '@chakra-ui/core'
import Icon from 'UI/Icon/Icon'
import styles from './styles'

// pseudo elements Chakra lacks
import sassStyles from './styles.module.scss'

const Header = ({ isMobileMenuOpen, onToggleMobileMenu = (f) => f, onSearchClick = (f) => f, onCheckoutClick = (f) => f, onAccountClick = (f) => f }) => {

    return (
        <Flex {...styles.header}>
            <Box
                className={sassStyles.menuBtn}
                {...styles.menuBtn}
                onClick={onToggleMobileMenu}
                data-active={isMobileMenuOpen}
                marginRight="1"
            >
                <Box {...styles.middleMenuBar}/>
            </Box>

            <Flex as="nav" {...styles.navMenu} marginRight="1">
                <Heading {...styles.navHeading}>Menu</Heading>
                <Button {...styles.navBtn} display="block">Contatti</Button>
                <Button {...styles.navBtn}>Prodotti</Button>
                <Button {...styles.navBtn}>Offerte</Button>
                <Button {...styles.navBtn}>Trattamenti</Button>
                <Button {...styles.navBtn}>Prenotazione</Button>
            </Flex>

            <Flex {...styles.logo} marginRight="1">
                <Heading {...styles.logoHeading}>Svetla Estetica</Heading>
                <Text {...styles.logoSubHeading}>Viale Natale Betelli, 51 - Dalmine (BG)</Text>
            </Flex>

            <Button marginRight="1" {...styles.iconButton} onClick={onSearchClick}><Icon icon="search" /></Button>
            <Button marginRight="1" {...styles.iconButton} {...styles.accountBtn} onClick={onAccountClick}><Icon icon="user" /></Button>
            <Button {...styles.iconButton} onClick={onCheckoutClick}><Icon icon="shopping-bag" /></Button>
        </Flex>
    )
}

export default Header