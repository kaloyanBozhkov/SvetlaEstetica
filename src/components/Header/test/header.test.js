import { shallow } from 'enzyme'
import React from 'react'

import Header from '../Header'

describe('testing Header component', () => {

    const mockProps = {
        onToggleMobileMenu: jest.fn((f) => f),
        isMobileMenuOpen: false,
        onAccountClick: jest.fn((f) => f),
        onCheckoutClick: jest.fn((f) => f),
        onSearchClick: jest.fn((f) => f)
    }

    beforeEach(() => {
        jest.clearAllMocks()
    })

    it('should render header with default props', () => {
        const wrapper = shallow(<Header />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should run toggle menu fn', () => {
        const wrapper = shallow(<Header {...mockProps} />)
        const menuBtn = wrapper.find('.menuBtn')
        expect(mockProps.onToggleMobileMenu).toHaveBeenCalledTimes(0)
        menuBtn.simulate('click')
        expect(mockProps.onToggleMobileMenu).toHaveBeenCalledTimes(1)
        menuBtn.simulate('click')
        expect(mockProps.onToggleMobileMenu).toHaveBeenCalledTimes(2)
    })


    // it('should run search fn', () => {
    //     const wrapper = shallow(<Header {...mockProps} />)
    //     const menuBtn = wrapper.find('Button')
    // })


    it('should change data active attribute of menu btn based on prop', () => {
        let wrapper = shallow(<Header {...mockProps} />)
        let menuBtn = wrapper.find('.menuBtn')
        expect(menuBtn.prop('data-active')).toEqual(false)
        wrapper = shallow(<Header {...mockProps} isMobileMenuOpen />)
        menuBtn = wrapper.find('.menuBtn')
        expect(menuBtn.prop('data-active')).toEqual(true)
    })
})