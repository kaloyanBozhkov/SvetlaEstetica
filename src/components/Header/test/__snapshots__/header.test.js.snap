// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header component should render header with default props 1`] = `
<Flex
  alignItems="center"
  backgroundColor="white"
  boxShadow="0px 0px 5px rgba(0, 0, 0, 0.25)"
  color="#4B4B4B"
  height="100%"
  justifyContent="space-between"
  paddingLeft={
    Object {
      "base": "50",
      "md": "8",
      "sm": "4",
      "xs": "2",
    }
  }
  paddingRight={
    Object {
      "base": "50",
      "md": "8",
      "sm": "4",
      "xs": "2",
    }
  }
  width="100%"
>
  <Box
    alignItems="center"
    className="menuBtn"
    cursor="pointer"
    display={
      Object {
        "base": "none",
        "xs": "flex",
      }
    }
    height="4"
    marginRight="1"
    onClick={[Function]}
    order="-2"
    padding="2"
    position="relative"
    width="4"
  >
    <Box
      bg="background.800"
      height="1px"
      width="100%"
    />
  </Box>
  <Flex
    alignItems={
      Object {
        "base": "unset",
        "xs": "flex-start",
      }
    }
    as="nav"
    bg={
      Object {
        "base": "unset",
        "xs": "white",
      }
    }
    borderColor="background.200"
    borderRight={
      Object {
        "base": "none",
        "xs": "1px solid",
      }
    }
    bottom="0"
    flexDirection={
      Object {
        "base": "row",
        "xs": "column",
      }
    }
    fontSize="sm"
    height={
      Object {
        "base": "100%",
        "xs": "100vh",
      }
    }
    justifyContent={
      Object {
        "base": "center",
        "xs": "flex-start",
      }
    }
    left="0"
    marginRight="1"
    position={
      Object {
        "base": "relative",
        "xs": "fixed",
      }
    }
    top="0"
    transform={
      Object {
        "base": "none",
        "xs": "translateX(-100%)",
      }
    }
    width={
      Object {
        "base": "auto",
        "xs": "30rem",
      }
    }
    zIndex={
      Object {
        "base": "base",
        "xs": "hide",
      }
    }
  >
    <Heading
      alignItems="center"
      alignSelf="stretch"
      borderBottom="1px solid"
      borderColor="background.200"
      display={
        Object {
          "base": "none",
          "xs": "flex",
        }
      }
      fontSize="tighter"
      height="24"
      justifyContent="center"
      marginBottom="5"
      marginLeft="2"
      marginRight="2"
    >
      Menu
    </Heading>
    <Button
      _hover={
        Object {
          "bg": "unset",
        }
      }
      bg="unset"
      display="block"
      fontSize="sm"
      fontWeight={
        Object {
          "base": "bold",
          "xs": "normal",
        }
      }
      height={
        Object {
          "base": "100%",
          "xs": "4rem",
        }
      }
      marginBottom={
        Object {
          "base": "0",
          "xs": "2",
        }
      }
      p={
        Object {
          "base": "0",
        }
      }
      textAlign={
        Object {
          "base": "left",
          "xs": "center",
        }
      }
      width={
        Object {
          "base": "13",
          "xs": "100%",
        }
      }
    >
      Contatti
    </Button>
    <Button
      _hover={
        Object {
          "bg": "unset",
        }
      }
      bg="unset"
      display={
        Object {
          "base": "none",
          "xxs": "block",
        }
      }
      fontSize="sm"
      fontWeight={
        Object {
          "base": "bold",
          "xs": "normal",
        }
      }
      height={
        Object {
          "base": "100%",
          "xs": "4rem",
        }
      }
      marginBottom={
        Object {
          "base": "0",
          "xs": "2",
        }
      }
      p={
        Object {
          "base": "0",
        }
      }
      textAlign={
        Object {
          "base": "left",
          "xs": "center",
        }
      }
      width={
        Object {
          "base": "13",
          "xs": "100%",
        }
      }
    >
      Prodotti
    </Button>
    <Button
      _hover={
        Object {
          "bg": "unset",
        }
      }
      bg="unset"
      display={
        Object {
          "base": "none",
          "xxs": "block",
        }
      }
      fontSize="sm"
      fontWeight={
        Object {
          "base": "bold",
          "xs": "normal",
        }
      }
      height={
        Object {
          "base": "100%",
          "xs": "4rem",
        }
      }
      marginBottom={
        Object {
          "base": "0",
          "xs": "2",
        }
      }
      p={
        Object {
          "base": "0",
        }
      }
      textAlign={
        Object {
          "base": "left",
          "xs": "center",
        }
      }
      width={
        Object {
          "base": "13",
          "xs": "100%",
        }
      }
    >
      Offerte
    </Button>
    <Button
      _hover={
        Object {
          "bg": "unset",
        }
      }
      bg="unset"
      display={
        Object {
          "base": "none",
          "xxs": "block",
        }
      }
      fontSize="sm"
      fontWeight={
        Object {
          "base": "bold",
          "xs": "normal",
        }
      }
      height={
        Object {
          "base": "100%",
          "xs": "4rem",
        }
      }
      marginBottom={
        Object {
          "base": "0",
          "xs": "2",
        }
      }
      p={
        Object {
          "base": "0",
        }
      }
      textAlign={
        Object {
          "base": "left",
          "xs": "center",
        }
      }
      width={
        Object {
          "base": "13",
          "xs": "100%",
        }
      }
    >
      Trattamenti
    </Button>
    <Button
      _hover={
        Object {
          "bg": "unset",
        }
      }
      bg="unset"
      display={
        Object {
          "base": "none",
          "xxs": "block",
        }
      }
      fontSize="sm"
      fontWeight={
        Object {
          "base": "bold",
          "xs": "normal",
        }
      }
      height={
        Object {
          "base": "100%",
          "xs": "4rem",
        }
      }
      marginBottom={
        Object {
          "base": "0",
          "xs": "2",
        }
      }
      p={
        Object {
          "base": "0",
        }
      }
      textAlign={
        Object {
          "base": "left",
          "xs": "center",
        }
      }
      width={
        Object {
          "base": "13",
          "xs": "100%",
        }
      }
    >
      Prenotazione
    </Button>
  </Flex>
  <Flex
    alignItems="center"
    flex="1"
    flexDirection="column"
    height="6xl"
    justifyContent="center"
    marginRight="1"
  >
    <Heading
      fontSize={
        Object {
          "mdX": "xl",
          "xs": "lg",
        }
      }
      fontWeight="medium"
      lineHeight={
        Object {
          "md": "wider",
          "xs": "tighter",
        }
      }
    >
      Svetla Estetica
    </Heading>
    <Text
      color="text.400"
      fontSize={
        Object {
          "md": "xs",
          "mdX": "md",
          "xs": "xxs",
        }
      }
    >
      Viale Natale Betelli, 51 - Dalmine (BG)
    </Text>
  </Flex>
  <Button
    _hover={
      Object {
        "bg": "unset",
      }
    }
    bg="unset"
    fontSize={
      Object {
        "lg": "lg",
        "md": "md",
        "xs": "sm",
      }
    }
    height={
      Object {
        "md": "4",
        "xs": "3",
      }
    }
    marginRight="1"
    onClick={[Function]}
    width={
      Object {
        "md": "4",
        "xs": "3",
      }
    }
  >
    <Icon
      icon="search"
    />
  </Button>
  <Button
    _hover={
      Object {
        "bg": "unset",
      }
    }
    bg="unset"
    fontSize={
      Object {
        "lg": "lg",
        "md": "md",
        "xs": "sm",
      }
    }
    height={
      Object {
        "md": "4",
        "xs": "3",
      }
    }
    marginRight="1"
    onClick={[Function]}
    order={
      Object {
        "base": "0",
        "xs": "-1",
      }
    }
    width={
      Object {
        "md": "4",
        "xs": "3",
      }
    }
  >
    <Icon
      icon="user"
    />
  </Button>
  <Button
    _hover={
      Object {
        "bg": "unset",
      }
    }
    bg="unset"
    fontSize={
      Object {
        "lg": "lg",
        "md": "md",
        "xs": "sm",
      }
    }
    height={
      Object {
        "md": "4",
        "xs": "3",
      }
    }
    onClick={[Function]}
    width={
      Object {
        "md": "4",
        "xs": "3",
      }
    }
  >
    <Icon
      icon="shopping-bag"
    />
  </Button>
</Flex>
`;
